pkg/controllers/leaderworkerset_controller.go

```diff
@@ -277,8 +300,32 @@ func (r *LeaderWorkerSetReconciler) rollingUpdateParameters(ctx context.Context,
 func (r *LeaderWorkerSetReconciler) SSAWithStatefulset(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet, partition, replicas int32) error {
 	log := ctrl.LoggerFrom(ctx)
 
+	// templateHash is not a reliable way to determine whether or not an lws object has been updated as seen in
+	// https://github.com/kubernetes-sigs/lws/issues/281
+	// If a leader sts already exists, but the template has not been updated, the templateHash of the leader is
+	// used to keep consistency in cases where two different templateHashes are calculated from the same LWS object
+	stsExists, sts, err := stsCreated(ctx, r.Client, lws)
+	if err != nil {
+		return err
+	}
+	templateHash := utils.LeaderWorkerTemplateHash(lws)
+	if stsExists {
+		templateUpdated, err := templateUpdated(ctx, r.Client, sts, lws)
+		if err != nil {
+			return err
+		}
+		if !templateUpdated {
+			templateHash = sts.Labels[leaderworkerset.TemplateRevisionHashKey]
+		}
+	}
```

Comment:
This should be called in `Reconcile` since multiple functions need to know the templateHash and whether the lws was updated.